lib/cocoapods/generate/podfile_generator.rb

```diff
@@ -140,26 +140,30 @@ def podfile_for_spec(spec)
         end
       end
 
-      def transitive_local_dependencies(spec, paths)
-        dependencies = spec.dependencies
-        return_list = []
-        dependencies.each do |dependency|
+      def transitive_local_dependencies(spec, paths, found_podspecs: {}, include_non_library_subspecs: true)
+        if include_non_library_subspecs
+          non_library_specs = spec.recursive_subspecs.select do |ss|
+            ss.test_specification? || (ss.respond_to?(:app_specification?) && ss.app_specification?)
+          end
+          non_library_specs.each do |subspec|
+            transitive_local_dependencies(subspec, paths, found_podspecs: found_podspecs, include_non_library_subspecs: false)
```

Comment:
I wasn't sure why we needed include_non_library_subspecs at  first, but confirmed with Sam. The recursive_subspecs method returns *all* descendant specs of the given spec. We specify include_non_library_subspecs as false, to avoid revisiting them.